### PR TITLE
Upgraded Mongodb Dependency from 2.2.35 to 3.6.8

### PR DESCRIPTION
**Pull request by Gauravp-NEC**

### CODE REVIEW

File: lib/database/sthDatabase.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ function getJSONCSVOptions(attrName) {
 function connect(params, callback) {
     connectionURL =
         // prettier-ignore
-        'mongodb://' + params.authentication + '@' + params.dbURI + '/' + params.database +
+        'mongodb://' + params.authentication + params.dbURI + '/' + params.database +
```
Comment (AlvaroVega) [inline lib/database/sthDatabase.js:93]: According with: https://docs.mongodb.com/manual/reference/connection-string/ when auth is provided '@' should be used

**Comment by fgalan** [comment]
@Gauravp-NEC This PR seems to be pretty similar to PR https://github.com/telefonicaid/fiware-sth-comet/pull/548. Or is there any difference?

**Comment by fgalan** [comment]
Similarly to what happened in PR https://github.com/telefonicaid/fiware-sth-comet/pull/548, unit tests are failing (see https://github.com/telefonicaid/fiware-sth-comet/runs/4481181797?check_suite_focus=true).

Thus, it seems is not a matter of just changing the version number in packages.json :) Probably something has changed in the module programming interface and the code should be fixed to support that.

**Comment by Gauravp-NEC** [comment]
> @Gauravp-NEC This PR seems to be pretty similar to PR #548. Or is there any difference?

@fgalan , I have removed the `@` from the connectionURL and connection is successfully established with the MongoDB with STH server running.

**Comment by fgalan** [comment]
> > @Gauravp-NEC This PR seems to be pretty similar to PR #548. Or is there any difference?
> 
> @fgalan , I have removed the `@` from the connectionURL and connection is successfully established with the MongoDB database with STH server is running. I have `node version 14.18.2` and `npm version 8.2.0` in my environment.

Could you run `npm test` and provide the output you get, please?

**Comment by Gauravp-NEC** [comment]
> > > @Gauravp-NEC This PR seems to be pretty similar to PR #548. Or is there any difference?
> > 
> > 
> > @fgalan , I have removed the `@` from the connectionURL and connection is successfully established with the MongoDB database with STH server is running. I have `node version 14.18.2` and `npm version 8.2.0` in my environment.
> 
> Could you run `npm test` and provide the output you get, please?

@fgalan , I have run the `npm test` and output is attached below.
[output.txt](https://github.com/telefonicaid/fiware-sth-comet/files/7698176/output.txt)

File: lib/database/sthDatabase.js (hunk 1 above)
Comment (AlvaroVega) [reply lib/database/sthDatabase.js:93]: A fix to allow that was done in PR https://github.com/telefonicaid/fiware-sth-comet/pull/564

**Comment by AlvaroVega** [comment]
IMHO we can close this PR

**Comment by AlvaroVega** [comment]
This branch should be syncronized with master branch, maybe that way will solve the issues

**Comment by Gauravp-NEC** [comment]
> This branch should be synchronized with master branch, maybe that way will solve the issues

@AlvaroVega , Did you mean this PR should get merged to master?

**Comment by fgalan** [comment]
>Did you mean this PR should get merged to master?

In fact, it's the other way around :) To merge master *into* your PR branch.

This can be easily done using "Resolve conflics" buttom below and accepting the master branch alternative in each conflict.

**Comment by Gauravp-NEC** [comment]
> > Did you mean this PR should get merged to master?
> 
> In fact, it's the other way around :) To merge master _into_ your PR branch.
> 
> This can be easily done using "Resolve conflics" buttom below and accepting the master branch alternative in each conflict.

@fgalan , Conflict is resolved now.

**Comment by fgalan** [comment]
> > > Did you mean this PR should get merged to master?
> > 
> > 
> > In fact, it's the other way around :) To merge master _into_ your PR branch.
> > This can be easily done using "Resolve conflics" buttom below and accepting the master branch alternative in each conflict.
> 
> @fgalan , Conflict is resolved now.

Yes, conflict is solved and master merged into this PR branch. However, not the test fail persists:

imagen

Currently @AlvaroVega is having a look to this and doing progress in PR https://github.com/telefonicaid/fiware-sth-comet/pull/568. If you want to help, please have a look to it and provide feedback that could help to solve the problem with the test fail.

**Comment by fgalan** [comment]
Overpassed by PR https://github.com/telefonicaid/fiware-sth-comet/pull/568

**Pull request closed without merging**